File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums)
        print(freq)
        sorted_keys = sorted(freq.keys(), key=lambda x:freq[x], reverse=True)

        print(sorted_keys)

        return sorted_keys[:k]

File: Data Structures & Algorithms/search-insert-position/submission-4.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size()-1;

        while(l<=r){
            if(l==r && nums[l] != target){
                if(nums[l] > target){
                    return l;
                }
                else{
                    return l+1;
                }
                
            }

            int mid = (r-l)/2+l;

           
            if(target > nums[mid]) l = mid+1;
            else if(target < nums[mid]) r = mid;
            else{
                return mid;
            }

            

        }
        return 0;
    }
};